core.js

```diff
@@ -14,6 +13,11 @@ function makeRequest (object, callback, responseCB) {
         method: object.method || 'GET'
     };
 
+    if (options.hostname === 'jenkins.whoop.com') {
+        options.auth = process.env.GITHUB_USERNAME + ':' +
+        process.env.GITHUB_API_TOKEN;
+    }
```

Comment:
This looks very similar to the conditional below it. Why not change to `if (options.hostname === 'api.github.com' || options.hostname === 'jenkins.whoop.com') {` ?

You might also write a function `getAuthByHostname(options.hostname)` that returns either an appropriate string for `options.auth` or `undefined`.